agent/connect/ca/provider_vault.go

```diff
@@ -922,6 +921,13 @@ func vaultLogin(client *vaultapi.Client, authMethod *structs.VaultAuthMethod) (*
 	return resp, nil
 }
 
+// Note the authMethod's parameters (Params) is populated from a freeform map
+// in the configuration where they could hardcode values to be passed directly
+// to the `auth/*/login` endpoint. So each auth method's authentication code
+// needs to handle both these cases. The legacy case (which should be
+// deprecated) where the user has hardcoded login values directly (eg. a `jwt`
```

Comment:
Do you think that we should officially deprecate the direct `auth/<method>/login` configuration support? If so we should probably add an issue to our backlog.

Reply:
Added to our backlog.